### PR TITLE
[SPARK-12115] [SparkR] Change numPartitions() to getNumPartitions() to be consistent with Scala/Python

### PR DESCRIPTION
**Pull request by yanboliang**
Change `numPartitions()` to `getNumPartitions()` to be consistent with Scala/Python.
<del>Note: If we can not catch up with 1.6 release, it will be breaking change for 1.7 that we also need to explain in release note.<del>

cc @sun-rui @felixcheung @shivaram

### CODE REVIEW

**Comment by felixcheung** [comment]
This is actually not exported from SparkR - since it is first integrated in Spark 1.4, SparkR is exporting a smaller/different set of API.
You can see in https://github.com/apache/spark/blob/master/R/pkg/NAMESPACE
 
While it is possible to access this with Spark:::numPartitions(), it has been available since Spark 1.4 so this rename is actually going to be a breaking change (of an internal API).
So I'd vote for no change in SparkR.

**Comment by srowen** [comment]
Yeah, it's unfortunately because until a change 3 days ago, we had 3 different methods in 4 languages for this simple function. Now everything but R uses `getNumPartitions`. It's not worth breaking something but may be worth adding/deprecating a method.

**Comment by yanboliang** [comment]
@felixcheung Thanks for your comments. I got it's not an exposed API, but I think to provide consistent function name is necessary especially when we want to expose RDD API someday. I think another solution is to add ```getNumPartitions``` as an alias of ```numPartitions``` which will not cause breaking change, and we can expose ```getNumPartitions``` when we want to expose RDD API. Looking forward to other members comments.

**Comment by yanboliang** [comment]
@srowen I think we have the consistent idea that to provide both ```getNumPartitions``` and ```numPartitions``` at SparkR side, and mark ```numPartitions``` as deprecated.

**Comment by sun-rui** [comment]
+1 @yanboliang

**Comment by shivaram** [comment]
Yeah this looks fine. Since it wasn't a publicly exposed API I don't think the backwards compatibility matters that much -- but for right now I'm +1 on just deprecating `numPartitions`.

File: R/pkg/R/RDD.R [hunk 1]
```diff
@@ -306,17 +306,26 @@ setMethod("checkpoint",
...
+#' @rdname getNumPartitions
+#' @aliases getNumPartitions,RDD-method
+#' @noRd
+setMethod("getNumPartitions",
+          signature(x = "RDD"),
+          function(x) {
+            callJMethod(getJRDD(x), "getNumPartitions")
+          })
+
+#' numPartitions is the same as getNumPartitions
```
Comment (shivaram) [inline R/pkg/R/RDD.R]: Could you add a note that this is deprecated here ?

Also this is the title of the document, so its better to have this be something like `Gets the number of partitions of an RDD` cc @felixcheung

Comment (felixcheung) [inline R/pkg/R/RDD.R]: It is, although we are not generating doc for internal API via `@noRd` below...

Comment (felixcheung) [inline R/pkg/R/RDD.R]: In fact, if we are deprecating, consider adding
```
.Deprecated(newFuncSig, old = oldFuncSig)
```
call to the deprecated version

**Comment by shivaram** [comment]
LGTM. Merging this. Thanks @yanboliang

**Pull request closed without merging**